code/__HELPERS/plist.dm

```diff
@@ -0,0 +1,126 @@
+// PLISTS BY RPAUL
+// basically a better version of lists mimicing php lists
+// http://www.byond.com/docs/ref/info.html#/list/operators del this later
+// http://www.byond.com/docs/ref/info.html#/operator/overload this too
+
+plist
+    var/list/_list = list()
+
+/*
+=================
+    OPERATORS
+=================
+*/
+//return at idx TODO maybe sanitize. list["[var]" fix that somehow
+plist/proc/operator[](var/idx)
+    return _list[idx]
+
+//check if list needs to be extended, so if idx is num and > then len. if idx is str we dont care
+//if idx is null, append
+plist/proc/operator[]=(idx, B)
+    _list.Insert(idx, B)
```

Comment:
Why isn't this just `_list[idx] = B`? Inserting doesn't really make any sense here

Reply:
insert features appending if idx = 0, thats why im using that here

Reply:
dunno if the same thing happens when you call list[0]

Reply:
Yes but its effect is completely different
It *inserts* an element, not changes an existing one